fe-code-review/code.js

```diff
@@ -0,0 +1,142 @@
+                            replyTo: config.projects[project].email.replyTo,
+                            subject: 'Email subject',
+                            templateId: config.projects[project].email.template.collector,
+                            params: {
+                                downloadUrl: `https://url.go/download?requestKey=${requestKey}&hash=${hash}`,
+                                uploadUrl: `https://url.go/upload?requestKey=${requestKey}&hash=${hash}`,
+                                confirmUrl: `https://url.go/confirm?requestKey=${requestKey}&hash=${hash}`
+                            },
+                            tags: ['request'],
+                            to: [{ email: collectorEmail , name: collectorName }],
+                        };
+                        logDebug('Send config:', sendConfig);
+
+                        try {
+                            await db.setEmailLog({collectorEmail, idCollector, idUser, requestKey})
+                        } catch (e) {
+                            logDebug('extract() setEmailLog error=', e);
+                        }
+
+                        /* send email */
+                        const resp = await email.send(sendConfig, config.projects[project].email.apiKey);
+                        logDebug('extract() resp=', resp);
+
+                        // update DB with result
+                        await db.setUserCollectorRequestKeyRes(requestKey, idUser, idCollector, resp);
+
+                        if (!sentStatus[collectorName])
+                            sentStatus[collectorName] = {};
+                        sentStatus[collectorName][collectorEmail] = resp;
+
+                        if (!resp) {
+                            logError('extract() Sending email failed: ', resp);
+                        }
+                    }
+                }
+                await db.updateStatus(requestID, 100, '');
+
+                logDebug('FINAL SENT STATUS:');
+                console.dir(sentStatus, {depth: null});
+
+                //if (!allSent)
+                //return res.status(500).json({requestID, message: 'Failed sending email'});
+
+                await db.updateStatus(requestID, 500, '');
+
+                /* prepare summary email */
+                const summaryConfig = {
+                    //bcc: [{ email: 'unknown@domain.com', name: 'Tomas' }],
+                    sender: config.projects[project].email.sender,
+                    replyTo: config.projects[project].email.replyTo,
+                    subject: 'Oppsummering Kravsforespørsel',
+                    templateId: config.projects[project].email.template.summary,
+                    params: {
+                        collectors: sentStatus,
+                    },
+                    tags: ['summary'],
+                    to: [{ email: 'unknown@otherdomain.no' , name: 'Tomas' }], // FIXXX: config.projects[project].email.sender
```

Comment:
Do email and name stay one and the same all the time? And the data is sensitive, it can't be exposed in code